apps/client/src/pages/dashboard/components/sections/BookmarkSection.tsx

```diff
@@ -33,75 +35,103 @@ const BookmarkSection = ({
   onAllViewClick,
   isAllViewExpanded = false,
 }: BookmarkSectionProps) => {
-  // 현재 카테고리의 북마크만 필터링
   const filteredBookmarks =
     activeCategory === UI_TEXT.category.all
       ? bookmarks
-      : bookmarks.filter(() => {
-          // 실제로는 카테고리별 필터링 로직이 필요하지만,
-          // 현재는 mock 데이터이므로 간단히 처리
-          return true;
-        });
+      : activeCategory === '안 읽은 정보'
+        ? bookmarks.filter((bookmark) => !bookmark.isRead)
+        : bookmarks.filter((bookmark) => {
+            const category = categories.find(
+              (cat) => cat.text === activeCategory
+            );
+            return category && bookmark.categoryId === category.id;
+          });
+
+  const filteredCount = filteredBookmarks.length;
+
+  const showAllViewBtn = filteredCount > 12;
+
+  const getEmptyStatePreset = () => {
+    if (bookmarks.length === 0) {
+      return 'url';
+    }
+    if (activeCategory === '안 읽은 정보') {
+      const unreadCount = bookmarks.filter(
+        (bookmark) => !bookmark.isRead
+      ).length;
+      if (unreadCount === 0) {
+        return 'unread';
+      }
+    }
+    return 'unread';
+  };
```

Comment:
지금 코드도 좋지만 조건에 네이밍을 붙이고, `filter`로 걸러주는 로직을 `some`으로 아래와 같이 작성해보는 것은 어떤가요?
테스트 해보시고 괜찮다면 한번 고려할만한 예시 코드가 될 것 같아요 👍 

```suggestion
const getEmptyStatePreset = () => {
  if (bookmarks.length === 0) return 'url';

  const isUnreadCategory = activeCategory === '안 읽은 정보';
  const hasUnread = bookmarks.some((bookmark) => !bookmark.isRead);

  if (isUnreadCategory && !hasUnread) {
    return 'unread';
  }

  return 'unread';
};
```

Reply:
너무 좋은 방법이네요!! 적용해보니 조건 네이밍 덕분에 가독성도 좋아지고, some 활용도 직관적이라 확실히 더 깔끔한 코드가 된 것 같아요. 리뷰 감사합니다!!